### ISSUE TITLE
Is there any reason to use $map in this module vs the result?

### ISSUE THREAD

**Issue by christianbundy** [dominictarr/map-filter-reduce#7]
I feel pretty comfortable with mapping with pull-streams, but I noticed that this module provides the same functionality. Is it any faster to use `$map` or should I just do it in my pull-stream?

**Comment by dominictarr** [comment]
I don't know you should measure it!

But theoretically, there are two reasons you'd use $map here, 1) you have MFR running on the server, so maybe you want to do the map there instead of the client, or 2) because to prepare the data for the $reduce step. it's probably a bit faster to make the map pure js, rather than basically an interpreted mini language. on the other hand, using map may save you serialization/deserialization by sending less data.

**Comment by christianbundy** [comment]
True! Those are good points, thanks for helping me understand this a bit better.
